petab/v2/lint.py

```diff
@@ -0,0 +1,549 @@
+                    | (
+                        pd.api.types.is_string_dtype(df[ESTIMATE])
+                        and df[ESTIMATE] != "1"
+                    )
+                ]
+            )
+            if non_estimated_par_ids:
+                if NOMINAL_VALUE not in df:
+                    return ValidationError(
+                        "Parameter table contains parameters "
+                        f"{non_estimated_par_ids} that are not "
+                        "specified to be estimated, "
+                        f"but column {NOMINAL_VALUE} is missing."
+                    )
+                try:
+                    df.loc[non_estimated_par_ids, NOMINAL_VALUE].apply(float)
+                except ValueError:
+                    return ValidationError(
+                        f"Expected numeric values for `{NOMINAL_VALUE}` "
+                        "in parameter table "
+                        "for all non-estimated parameters."
+                    )
+
+            assert_parameter_id_is_string(df)
+            assert_parameter_scale_is_valid(df)
+            assert_parameter_bounds_are_numeric(df)
+            assert_parameter_estimate_is_boolean(df)
+            assert_unique_parameter_ids(df)
+            check_parameter_bounds(df)
+            assert_parameter_prior_type_is_valid(df)
+            assert_parameter_prior_parameters_are_valid(df)
+
+        except AssertionError as e:
+            return ValidationError(str(e))
+
+
+class CheckAllParametersPresentInParameterTable(ValidationTask):
+    """Ensure all required parameters are contained in the parameter table
+    with no additional ones."""
+
+    def run(self, problem: Problem) -> ValidationIssue | None:
+        if (
+            problem.model is None
+            or problem.parameter_df is None
+            or problem.observable_df is None
+            or problem.measurement_df is None
+        ):
+            return
+
+        required = get_required_parameters_for_parameter_table(problem)
+
+        allowed = get_valid_parameters_for_parameter_table(
+            model=problem.model,
+            condition_df=problem.condition_df,
+            observable_df=problem.observable_df,
+            measurement_df=problem.measurement_df,
+            mapping_df=problem.mapping_df,
+        )
+
+        actual = set(problem.parameter_df.index)
+        missing = required - actual
+        extraneous = actual - allowed
+
+        # missing parameters might be present under a different name based on
+        # the mapping table
+        if missing and problem.mapping_df is not None:
+            model_to_petab_mapping = {}
+            for map_from, map_to in zip(
+                problem.mapping_df.index.values,
+                problem.mapping_df[MODEL_ENTITY_ID],
+                strict=True,
+            ):
+                if map_to in model_to_petab_mapping:
+                    model_to_petab_mapping[map_to].append(map_from)
```

Comment:
Why would multiple model IDs map to a single PEtab ID? Is it e.g. to observe the sum of all species regardless of isotope state in rule-based modeling? Then I would have more questions.. but irrelevant here.

Reply:
Iirc, the only reason is that so far, it's not explicitly forbidden.